services/QuillLMS/app/models/concerns/teacher.rb

```diff
@@ -32,10 +32,6 @@ def has_classrooms?
     classrooms_i_teach.any? && !classrooms_i_teach.all?(&:new_record?)
   end
 
-  def classrooms_i_teach
```

Comment:
I suspect (and Dan can confirm), that the raw SQL used in the join here might mess up the auto caching built into Rails that we're trying to use here.  That said, I agree with Emilia that this is used extensively, so if that IS the problem, we probably just have to refactor this function to work with the caching system we want to use here.

Reply:
I'll put this back in since that removal is unrelated to this PR. 

I removed this because `classrooms_i_teach` is actually defined in two places (it's also [defined on the User model as an association](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/app/models/user.rb#L98)). I thiiink that other definition overwrites this one since Teacher is a mixin to User (and it appears above the association), but I'm not 100% sure.

I'm going to write up a card to address this (and not do it in this PR), since we should only have one definition.

Reply:
just want to drop a note that the share activity/activity pack feature that will be released tomorrow utilizes `classrooms_i_teach` but via the User model association